File: ejercicios/python/ejercicio088.py
from random import randint

__size = 5

def llenado_aleatorio(m):
    for i in range(__size):
        m.append([randint(1, 100) for i in range(__size)])

def mostrar_matriz(m):
    for i in m:
        for j in i:
            if(j < 10):
                print("  " + str(j), end='\t')
            else:
                print(" " + str(j), end='\t')
        print("")
        
def ordenar_matriz(m):
    for i in range(__size):
        for j in range(__size):
            for ia in range(__size):
                for ja in range(__size):
                    if(m[i][j] < m[ia][ja]):
                        temp = m[i][j]
                        m[i][j] = m[ia][ja]
                        m[ia][ja] = temp
        
if __name__ == "__main__":
    matriz=[]
    llenado_aleatorio(matriz)
    print("matriz original")
    mostrar_matriz(matriz)
    ordenar_matriz(matriz)
    print("matriz ordenada")
    mostrar_matriz(matriz)

File: ejercicios/c++/ejercicio088.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

//procedimiento que llena aleatriamente la matriz
void llenado_aleatorio(int[5][5]);

//procedimiento que muestra en pantalla la matriz
void ver_matriz(int[5][5]);

//procedimiento que ordena la matriz
void ordenar_matriz(int[5][5]);

int main(int argc, char *argv[]) {
	int matriz[5][5];
	
	srand(time(0));
	llenado_aleatorio(matriz);
	
	cout << "matriz original" << endl;
	ver_matriz(matriz);
	
	ordenar_matriz(matriz);
	
	cout << "matriz ordenada" << endl;
	ver_matriz(matriz);
	return 0;
}

void llenado_aleatorio(int m[5][5]){
	int aleatorio;
	for(int f=0; f<5; f++){
		for(int c=0; c<5; c++){
			do{
				aleatorio = 1 + rand() % 99;
			}while(aleatorio % 5 == 0);
			m[f][c] = aleatorio;
		}
		cout << endl;
	}
}
void ver_matriz(int m[5][5]){
	for(int f=0; f<5; f++){
		for(int c=0; c<5; c++){
			if(m[f][c] < 10)
				cout << " " << m[f][c] << "\t";
			else
				cout << m[f][c] << "\t";
		}
		cout << endl;
	}
}
void ordenar_matriz(int m[5][5]){
	int temp;
	for(int f=0; f<5; f++){
		for(int c=0; c<5; c++){
			for(int fa=0; fa<5; fa++){
				for(int ca=0; ca<5; ca++){
					if(m[f][c] < m[fa][ca]){
						temp = m[f][c];
						m[f][c] = m[fa][ca];
						m[fa][ca] = temp;
					}
				}
			}
		}
	}
}
